src/CMakeLists.txt

```diff
@@ -8,6 +8,9 @@ set(UMF_SOURCES
     memory_provider_get_last_failed.c
     pool/pool_null.c
     pool/pool_trace.c
+    pool/disjoint/pool_disjoint.cpp
+    pool/disjoint/pool_disjoint_config_parser.cpp
+    pool/disjoint/pool_disjoint_heap_manager.cpp
     provider/provider_null.c
     provider/provider_trace.c
```

Comment:
Unrelated to this PR but why do we have pool/provider trace in the public APIs (include/pool and include/provider)? Those were supposed to be for testing only, I don't think we want to expose it to the user

Reply:
It was suggested by @bratpiorka . @bratpiorka Could you explain it to @igchor ?

Reply:
Ok we could move them to the test folder, but in the separate PR